includes/RestApi/SiteGenController.php

```diff
@@ -210,45 +191,43 @@ public function generate_sitegen_meta( \WP_REST_Request $request ) {
 	 * @return array
 	 */
 	public function get_homepages( \WP_REST_Request $request ) {
+		$existing_homepages = SiteGenService::get_homepages();
+		if ( ! empty( $existing_homepages ) ) {
+			return new \WP_REST_Response( $existing_homepages, 200 );
+		}
```

Comment:
Will this work? Cause I believe we do check in flow for `homepages > data` in flow before we make this call and won't this result in an unnecessary wp read?